packages/dev-server/src/HermesBundler.ts

```diff
@@ -9,19 +10,28 @@ import resolveFrom from 'resolve-from';
 
 export async function shouldBuildHermesBundleAsync(
   projectRoot: string,
+  target: ProjectTarget,
```

Comment:
let's skip switching behavior based on target and instead try to read from native project before app config, as described in comment below